code/drasil-lang/Language/Drasil/Space.hs

```diff
@@ -25,9 +25,20 @@ data Space =
   | Actor String
   | DiscreteD [Double]
   | DiscreteS [String] --ex. let Meal = {"breakfast", "lunch", "dinner"}
+  | Mapping [Space] Space -- TODO: this one shouldn't be exposed directly, only through a smart constructor
```

Comment:
So we might want to use `NonEmpty` instead of a raw list - we don't want 0-ary functions.